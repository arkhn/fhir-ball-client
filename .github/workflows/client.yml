name: pyrog-client

on: [push]

jobs:
  build:
    name: Build Pyrog client
    runs-on: ubuntu-18.04

    strategy:
      matrix:
        node-version: [12.x]

    steps:
      - uses: actions/checkout@v1

      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}

      - name: Cache node modules
        uses: actions/cache@v1
        with:
          path: client/node_modules
          key: ${{ runner.OS }}-${{ matrix.node-version}}-node-modules-${{ hashFiles('client/yarn.lock') }}

      - name: Install Dependencies
        run: |
          cd client
          yarn install

      - name: Build
        run: |
          cd client
          yarn build

  # Same cleaning need to be done, see Issue #105
  #      - name: Lint
  #        run: |
  #          cd client
  #          yarn lint

  publish:
    name: Build and publish
    env:
      GITHUB_REGISTRY: docker.pkg.github.com
      IMAGE: arkhn/pyrog-client
    runs-on: ubuntu-18.04
    needs: build
    if: github.ref == 'refs/heads/master'
    steps:
      - uses: actions/checkout@v1

      - name: Build
        run: |
          cd client/
          yarn docker:build:staging

      - name: Tag docker images
        run: |
          docker tag $IMAGE $IMAGE:$GITHUB_SHA
          docker tag $IMAGE $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:latest
          docker tag $IMAGE $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:$GITHUB_SHA

      - name: Publish to the docker hub
        run: |
          echo "${{ secrets.DOCKER_PASSWORD }}" |  docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
          docker push $IMAGE:latest
          docker push $IMAGE:$GITHUB_SHA

      - name: Publish to Github Packages
        run: |
          echo "${{ secrets.GITHUB_TOKEN }}" | docker login $GITHUB_REGISTRY -u ${{ github.actor }} --password-stdin
          docker push $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:latest
          docker push $GITHUB_REGISTRY/${{ github.repository }}/pyrog-client:$GITHUB_SHA
