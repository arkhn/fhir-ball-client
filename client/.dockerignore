# ignore everything
**/*

# except
!.dockerignore
!Dockerfile
!yarn.lock
!package.json
!public
!src
!tsconfig.json
!tsconfig.build.json
!.env*
!build