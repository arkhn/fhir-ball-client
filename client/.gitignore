node_modules/
dist/*
build/

coverage
yarn-error.log

.DS_Store