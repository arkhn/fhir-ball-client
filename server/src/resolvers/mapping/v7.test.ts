import { PrismaClient } from '@prisma/client'

import importMappingV7 from './v7'
import * as mappingV7 from '../../../test/fixtures/chimio-mapping-v7.json'

const mockCreateResource = jest.fn()
const mockFindManyUser = jest.fn(() => [{ email: 'admin@arkhn.com' }])
jest.mock('@prisma/client', () => ({
  PrismaClient: jest.fn().mockImplementation(() => ({
    resource: {
      create: (data: any) => mockCreateResource(data),
    },
    user: {
      findMany: mockFindManyUser,
    },
  })),
}))

describe('import mapping V7', () => {
  const sourceId = '01234567'
  const resourceCount = 2
  const { resources } = mappingV7 as any

  beforeEach(() => {
    mockCreateResource.mockClear()
  })

  it('should send a query per resource', async () => {
    await importMappingV7(new PrismaClient(), sourceId, resources)
    expect(mockCreateResource).toHaveBeenCalledTimes(resourceCount)
    expect(mockCreateResource.mock.calls[0]).toMatchSnapshot() // EpisodeOfCare - HopitalStay
    expect(mockCreateResource.mock.calls[1]).toMatchSnapshot() // Patient
  })

  it('should raise an error if importing mapping with unexisting comment author', async () => {
    mockFindManyUser.mockReturnValueOnce([{ email: 'user@arkhn.com' }])
    const t = importMappingV7(new PrismaClient(), sourceId, resources)
    expect(t).rejects.toThrowError(
      'trying to import a mapping with unexisting comment author',
    )
  })

  it('should have cleaned the resource and attributes', async () => {
    await importMappingV7(new PrismaClient(), sourceId, resources)
    expect(mockCreateResource.mock.calls[0][0]).toEqual({
      data: {
        id: resources[0].id,
        label: resources[0].label,
        primaryKeyOwner: resources[0].primaryKeyOwner,
        primaryKeyTable: resources[0].primaryKeyTable,
        primaryKeyColumn: resources[0].primaryKeyColumn,
        source: {
          connect: {
            id: sourceId,
          },
        },
        definitionId: 'EpisodeOfCare',
        attributes: {
          create: expect.arrayContaining([
            {
              id: expect.any(String),
              path: 'period.start',
              definitionId: 'dateTime',
              mergingScript: 'merge_concat',
              inputs: {
                create: expect.any(Array),
              },
            },
            {
              id: expect.any(String),
              path: 'managingOrganization.reference',
              comments: {
                create: [
                  {
                    author: {
                      connect: {
                        email: 'admin@arkhn.com',
                      },
                    },
                    content: 'test',
                    createdAt: '2020-04-02T08:53:07.997Z',
                  },
                ],
              },
              definitionId: 'string',
              mergingScript: null,
              inputs: {
                create: expect.any(Array),
              },
            },
          ]),
        },
        filters: {
          create: [
            {
              id: expect.any(String),
              value: '200',
              relation: '<=',
              sqlColumn: {
                create: {
                  id: expect.any(String),
                  column: 'CHAMBRE',
                  owner: 'OPS$ACHQ1ABC',
                  table: 'SEJOUR',
                },
              },
            },
          ],
        },
      },
    })
  })
})
