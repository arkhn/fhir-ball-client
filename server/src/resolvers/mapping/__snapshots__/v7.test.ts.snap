// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import mapping V7 should send a query per resource 1`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "definitionId": "string",
            "id": "ck8iiui3g0874euutj38hg2w3",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3g0878euut1905dqpz",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "LIBELTW",
                      "id": "ck8iiui3g0880euutb7uo17il",
                      "joins": Object {
                        "create": Array [
                          Object {
                            "id": "ck8iiui3g0882euutl4qtkhzn",
                            "tables": Object {
                              "create": Array [
                                Object {
                                  "column": "TWCLEUNIK",
                                  "id": "ck8iiui3g0884euutbugm7n1r",
                                  "table": "SEJOUR",
                                },
                                Object {
                                  "column": "TWCLEUNIK",
                                  "id": "ck8iiui3g0887euutfbeg2q1w",
                                  "table": "TYPESEJ",
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "table": "TYPESEJ",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].display",
          },
          Object {
            "definitionId": "Reference",
            "id": "ck8iiui3h0893euut2mt8id45",
            "mergingScript": null,
            "path": "patient",
          },
          Object {
            "definitionId": "uri",
            "id": "ck8iiui3h0895euut8qejc3lo",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0897euut1i0ztmbq",
                  "script": null,
                  "staticValue": "/Patient/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.system",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0900euutkj9nrxmn",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0902euutt8bfiib1",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "id": "ck8iiui3h0904euutow06q7fr",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.identifier.value",
          },
          Object {
            "definitionId": "Identifier",
            "id": "ck8iiui3h0908euutbfu6jg9b",
            "mergingScript": null,
            "path": "managingOrganization.identifier",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0909euutyv68l658",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0911euutqiqpvomq",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SERVICE",
                      "id": "ck8iiui3h0913euut4pe1okd6",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
          Object {
            "definitionId": "dateTime",
            "id": "ck8iiui3h0915euut8x2qfzks",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0918euutvumtyf8l",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEIN",
                      "id": "ck8iiui3h0920euutrnmfy23u",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0921euut6xx53utp",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREIN",
                      "id": "ck8iiui3h0923euut0o9d529y",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.start",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0927euutvs85rc77",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0929euutc5hvt0g8",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SECLEUNIK",
                      "id": "ck8iiui3h0931euut41jpxelx",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "definitionId": "code",
            "id": "ck8iiui3h0933euutk1aasv6p",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0935euutdt3suu39",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ETATSEJ",
                      "id": "ck8iiui3h0937euutgp9bkpn7",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "status",
          },
          Object {
            "definitionId": "CodeableConcept",
            "id": "ck8iiui3h0939euut7b7rn1ej",
            "mergingScript": null,
            "path": "type",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0941euutleuqj86l",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0944euutjgdyuh29",
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "patient.reference",
          },
          Object {
            "comments": Object {
              "create": Array [
                Object {
                  "author": Object {
                    "connect": Object {
                      "email": "admin@arkhn.com",
                    },
                  },
                  "content": "test",
                  "createdAt": "2020-04-02T08:53:07.997Z",
                },
              ],
            },
            "definitionId": "string",
            "id": "ck8iiui3h0945euuth5wnmtns",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0950euutatcsermm",
                  "script": null,
                  "staticValue": "internal",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.reference",
          },
          Object {
            "definitionId": "Period",
            "id": "ck8iiui3h0952euuttenzso2y",
            "mergingScript": null,
            "path": "period",
          },
          Object {
            "definitionId": "dateTime",
            "id": "ck8iiui3h0954euut8zwfbntz",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0956euutld7dnc4g",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "HEUREOUT",
                      "id": "ck8iiui3h0957euut2ymeffbb",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0959euut53yev1mp",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUT",
                      "id": "ck8iiui3h0961euutboxn52xy",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "period.end",
          },
          Object {
            "definitionId": "Coding",
            "id": "ck8iiui3h0964euuty3asc0h8",
            "mergingScript": null,
            "path": "type[0].coding",
          },
          Object {
            "definitionId": "Identifier",
            "id": "ck8iiui3h0965euutj05zb0ym",
            "mergingScript": null,
            "path": "identifier",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0967euut4u14vpqo",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0970euutcv8h5wls",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION",
                      "id": "ck8iiui3h0972euutm5dcshse",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0973euutvuifu98j",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADMISSION2",
                      "id": "ck8iiui3h0975euutx0l30vvb",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "select_first_not_empty",
            "path": "identifier[1].value",
          },
          Object {
            "definitionId": "code",
            "id": "ck8iiui3h0979euut1t9f8v80",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0980euutozr9mpeq",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TWCLEUNIK",
                      "id": "ck8iiui3h0982euutxt0rkzix",
                      "table": "SEJOUR",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "type[0].coding[0].code",
          },
          Object {
            "definitionId": "Identifier",
            "id": "ck8iiui3h0985euutjzzfkrdp",
            "mergingScript": null,
            "path": "patient.identifier",
          },
          Object {
            "definitionId": "Reference",
            "id": "ck8iiui3h0987euutru6fv2k9",
            "mergingScript": null,
            "path": "managingOrganization",
          },
          Object {
            "definitionId": "uri",
            "id": "ck8iiui3h0989euuta109k4fo",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0990euuttt00zl6b",
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
        ],
      },
      "definitionId": "EpisodeOfCare",
      "filters": Object {
        "create": Array [
          Object {
            "id": "ck8iiui3h0997euutcwndk9e7",
            "relation": "<=",
            "sqlColumn": Object {
              "create": Object {
                "column": "CHAMBRE",
                "id": "ck8iiui3h1000euutdbxfu1pc",
                "table": "SEJOUR",
              },
            },
            "value": "200",
          },
        ],
      },
      "id": "ck8iiui3g0872euut6adjvfnr",
      "label": "Hospital Stay",
      "primaryKeyColumn": "SECLEUNIK",
      "primaryKeyTable": "SEJOUR",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;

exports[`import mapping V7 should send a query per resource 2`] = `
Array [
  Object {
    "data": Object {
      "attributes": Object {
        "create": Array [
          Object {
            "definitionId": "Coding",
            "id": "ck8iiui3g0875euutkf6uodqk",
            "mergingScript": null,
            "path": "identifier[0].type.coding",
          },
          Object {
            "definitionId": "code",
            "id": "ck8iiui3g0876euutexvf69o0",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3g0877euuto7i0cdak",
                  "script": null,
                  "staticValue": "PI",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].code",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3g0879euut9967t9d4",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3g0881euuth0l41cz7",
                  "script": null,
                  "staticValue": "Patient internal identifier",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].display",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3g0883euut839w3p6a",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3g0885euutkmpy55f1",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DOSSIERPAT",
                      "id": "ck8iiui3g0886euutuxs3pf18",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].value",
          },
          Object {
            "definitionId": "Period",
            "id": "ck8iiui3g0888euutesfyyhsb",
            "mergingScript": null,
            "path": "identifier[2].period",
          },
          Object {
            "definitionId": "HumanName",
            "id": "ck8iiui3g0889euut5ggfp6q7",
            "mergingScript": null,
            "path": "name",
          },
          Object {
            "definitionId": "code",
            "id": "ck8iiui3h0890euut17jb4j1r",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0891euutd2ztrof3",
                  "script": null,
                  "staticValue": "maiden",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].use",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0892euut4600qeo2",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0894euut7no7sir4",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMJFPAT",
                      "id": "ck8iiui3h0896euutbs1keubp",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[1].family",
          },
          Object {
            "definitionId": "code",
            "id": "ck8iiui3h0898euut6a1dr77q",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0899euutaixgu3e2",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "SEXEPAT",
                      "id": "ck8iiui3h0901euutwnkvkkce",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "gender",
          },
          Object {
            "definitionId": "Address",
            "id": "ck8iiui3h0903euutkmr3v7ds",
            "mergingScript": null,
            "path": "address",
          },
          Object {
            "definitionId": "Reference",
            "id": "ck8iiui3h0905euutf90c2ltg",
            "mergingScript": null,
            "path": "managingOrganization",
          },
          Object {
            "definitionId": "uri",
            "id": "ck8iiui3h0906euutomx5t0cc",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0907euutgscl8qh8",
                  "script": null,
                  "staticValue": "/Organization/{id}",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.system",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0910euutnp9k1y22",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0912euutd49yrtfy",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CODESERV",
                      "id": "ck8iiui3h0914euutdsg7bypv",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.identifier.value",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0916euutdo1lqnj5",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0917euutopu6v5y7",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOOBSPAT",
                      "id": "ck8iiui3h0919euutq84e5y4s",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].value",
          },
          Object {
            "definitionId": "CodeableConcept",
            "id": "ck8iiui3h0922euutk1avxzu4",
            "mergingScript": null,
            "path": "identifier[0].type",
          },
          Object {
            "definitionId": "Identifier",
            "id": "ck8iiui3h0924euut1lrx4wpb",
            "mergingScript": null,
            "path": "identifier",
          },
          Object {
            "definitionId": "boolean",
            "id": "ck8iiui3h0925euut34vf7oot",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0926euutsbhh849y",
                  "script": null,
                  "staticValue": "false",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].userSelected",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0928euutab26dn9j",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0930euutoh4ziwrs",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PACLEUNIK",
                      "id": "ck8iiui3h0932euutxe8cyhzl",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[1].value",
          },
          Object {
            "definitionId": "dateTime",
            "id": "ck8iiui3h0934euutgylav4oi",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0936euutc7v7pazv",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINPAT",
                      "id": "ck8iiui3h0938euut659vuw8z",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.start",
          },
          Object {
            "definitionId": "dateTime",
            "id": "ck8iiui3h0940euutscrthm2f",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0942euut7uvubpg9",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEOUTPAT",
                      "id": "ck8iiui3h0943euutg93n40ko",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[2].period.end",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0946euutdz2z58cn",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0948euut96dbj7j0",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "NOMPAT",
                      "id": "ck8iiui3h0949euutvljkeug7",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].family",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0951euut7snrof9e",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0953euuthluoxzsl",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "TELPAT",
                      "id": "ck8iiui3h0955euutakdfx2xz",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "telecom[0].value",
          },
          Object {
            "definitionId": "date",
            "id": "ck8iiui3h0958euutnwve00s0",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0960euutgwqivaht",
                  "script": "clean_date",
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATENAISPA",
                      "id": "ck8iiui3h0962euutf0oxkj75",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "birthDate",
          },
          Object {
            "definitionId": "boolean",
            "id": "ck8iiui3h0963euute7h6ujpd",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0966euuta660ktzr",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "IDMOTIFINAC",
                      "id": "ck8iiui3h0968euut23dh97gd",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0969euutyjv4icnl",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "id": "ck8iiui3h0971euutj3twzt8e",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedBoolean",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0974euuteh2qez1w",
            "mergingScript": null,
            "path": "address[0].line",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0976euutj8u39lki",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0977euutajddkw84",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "VILLEPAT",
                      "id": "ck8iiui3h0978euut83bd52kx",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].city",
          },
          Object {
            "definitionId": "uri",
            "id": "ck8iiui3h0981euut49d3l3w9",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0983euutcuqi5vgm",
                  "script": null,
                  "staticValue": "http://terminology.interopsante.org/CodeSystem/v2-0203",
                },
              ],
            },
            "mergingScript": null,
            "path": "identifier[0].type.coding[0].system",
          },
          Object {
            "definitionId": "boolean",
            "id": "ck8iiui3h0984euut920wkr1a",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0986euuthbh3ieqk",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "INACTIVPAT",
                      "id": "ck8iiui3h0988euutxsvmfd7x",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "active",
          },
          Object {
            "definitionId": "code",
            "id": "ck8iiui3h0991euutjuo0d0iy",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0992euutkchrs1zr",
                  "script": null,
                  "staticValue": "official",
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].use",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0993euut7o6kldza",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h0994euutl8qzwjbn",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "PRENOMPAT",
                      "id": "ck8iiui3h0995euuteec5thl9",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "name[0].given[0]",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h0996euuttnkh9882",
            "mergingScript": null,
            "path": "name[0].given",
          },
          Object {
            "definitionId": "ContactPoint",
            "id": "ck8iiui3h0998euutg2ndrcq7",
            "mergingScript": null,
            "path": "telecom",
          },
          Object {
            "definitionId": "dateTime",
            "id": "ck8iiui3h0999euutzmj34tuj",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h1001euutxvy0picm",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "DATEINACPAT",
                      "id": "ck8iiui3h1002euutq3m8i7s4",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "deceasedDateTime",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3h1003euut9o2r3w8a",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3h1004euut66nveayc",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR3PAT",
                      "id": "ck8iiui3i1005euutkkjejufb",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3i1006euuta9wvm09q",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR1PAT",
                      "id": "ck8iiui3i1007euut6w476cku",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3i1008euut9o21ki3b",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "ADR2PAT",
                      "id": "ck8iiui3i1009euutn19nzf2n",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": "merge_concat",
            "path": "address[0].line[0]",
          },
          Object {
            "definitionId": "string",
            "id": "ck8iiui3i1010euutolkkbser",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3i1011euutapr9pdal",
                  "script": null,
                  "sqlValue": Object {
                    "create": Object {
                      "column": "CPOSTPAT",
                      "id": "ck8iiui3i1012euut314ev1fh",
                      "table": "PATIENT",
                    },
                  },
                  "staticValue": null,
                },
              ],
            },
            "mergingScript": null,
            "path": "address[0].postalCode",
          },
          Object {
            "definitionId": "uri",
            "id": "ck8iiui3i1013euutj23mrx86",
            "inputs": Object {
              "create": Array [
                Object {
                  "conceptMapId": null,
                  "id": "ck8iiui3i1014euut6ylie60l",
                  "script": null,
                  "staticValue": "Organization",
                },
              ],
            },
            "mergingScript": null,
            "path": "managingOrganization.type",
          },
          Object {
            "definitionId": "Identifier",
            "id": "ck8iiui3i1015euutgv9wqdv3",
            "mergingScript": null,
            "path": "managingOrganization.identifier",
          },
        ],
      },
      "definitionId": "Patient",
      "filters": Object {
        "create": Array [],
      },
      "id": "ck8iiui3g0873euutm08fnyrf",
      "label": "",
      "primaryKeyColumn": "PACLEUNIK",
      "primaryKeyTable": "PATIENT",
      "source": Object {
        "connect": Object {
          "id": "01234567",
        },
      },
    },
  },
]
`;
